crates/amalthea/src/socket/shell.rs

```diff
@@ -317,25 +265,30 @@ impl Shell {
         }
 
         // Process the comm open request
-        let result = self.open_comm(req.clone());
+        let result = self.open_comm(shell_handler, req.clone());
+
+        // There is no error reply for a comm open request. Instead we must send
+        // a `comm_close` message as soon as possible. The error is logged on our side.
+        if let Err(err) = result {
+            let reply = IOPubMessage::CommClose(CommClose {
+                comm_id: req.content.comm_id.clone(),
+            });
+            self.iopub_tx.send(reply).unwrap();
+            log::warn!("Failed to open comm: {err:?}");
```

Comment:
warn or error?

Reply:
Warn otherwise this shows up in the test output.